### PR TITLE
feat: adds send command to example client, currently non-functional

### PR DESCRIPTION
**Pull request by jowparks**
Tested with:
```
yarn dev send assetId=51f33a2f14f92735e562dc658a5639279ddca3d5079a6d1242b2a588a9cbf44c toPublicAddress=c016b357465f41fcfd896eb6b878bcd865726779096d208f64e54347df9b46c7 amount=1 memo=test fee=1
```

Submitted to network successfully:
```
Transaction: 258938cabd32124081a9fd9ed262d692cf94b4be79db3d59f148bc2927539cfb
Account: test
Status: confirmed
Type: send
Timestamp: 9/13/2023 5:05:37 PM PDT
Fee: $IRON 0.00000001
Block Hash: 00000000000000556c1fcc0a52849c00bd56b4cde647c7fc105efb5ac9bd9d9e
Block Sequence: 211968
Notes Count: 2
Spends Count: 1
Mints Count: 0
Burns Count: 0
Sender: xxx
```

### CODE REVIEW

File: example/src/send.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    { publicAddress: toPublicAddress },
+    BigInt(amount),
+    Buffer.from(assetId, "hex"),
+    BigInt(fee),
+    memo,
+  );
+  const posted = transaction.post(account.key.publicAddress, BigInt(fee));
+
+  console.log("Posted transaction:");
+  console.log(posted.toString("hex"));
```
Comment (jowparks) [inline example/src/send.ts:34]: Still need to call wallet-server api to post this, can add in follow up PR.

File: example/src/Client/utils/BlockProcessor.ts [hunk 2]
```diff
@@ -87,11 +89,11 @@ export class BlockProcessor {
 
     const batchSize = process.env["BLOCK_PROCESSING_BATCH_SIZE"]
       ? parseInt(process.env["BLOCK_PROCESSING_BATCH_SIZE"])
-      : 100;
-    for (let i = cachedHeadSequence; i < headSequence; i += batchSize) {
+      : 99;
+    for (let i = cachedHeadSequence + 1; i < headSequence; i += batchSize + 1) {
       await this._processBlockRange(i, Math.min(i + batchSize, headSequence));
```
Comment (jowparks) [inline example/src/Client/utils/BlockProcessor.ts:94]: main bug was here

File: example/src/Client/utils/AccountsManager.ts [hunk 3]
```diff
@@ -52,6 +53,7 @@ export class AccountsManager {
   /** Public key => AccountData */
   private accounts: Map<string, AccountData> = new Map();
   private events: EventEmitter = new EventEmitter();
+  private synced = false;
```
Comment (dgca) [inline example/src/Client/utils/AccountsManager.ts:56]: question: I don't see where this is getting set to true. Is this being used?

**Review by dgca — APPROVED** [review]

File: example/src/Client/utils/AccountsManager.ts (hunk 3 above)
Comment (jowparks) [reply example/src/Client/utils/AccountsManager.ts:56]: https://github.com/iron-fish/wallet-server/pull/27/files/a0948bb73d40f42c1d0e28a804d9f4d029705a39#diff-100d016004f580d38d2fcdcc9afbc0c4fe0693099a7e78ce238a0c66ba6696d0R93

**Pull request merged**